frontend/src/pages/quiz/tasks/[[...filters]].tsx

```diff
@@ -63,37 +75,60 @@ const Tasks: FC<TasksProps> = ({ applyingBenefits, beforeRetiring, filters, rece
               {t('restart-quiz')}
             </Button>
           </div>
-          <p>Answers:</p>
-          <ul className="mb-4 list-disc space-y-2 pl-7">
-            {filters?.answers?.map((answer) => (
-              <li key={answer}>{answer}</li>
-            ))}
-          </ul>
-          <p>Tags:</p>
-          <ul className="list-disc space-y-2 pl-7">
-            {filters?.tags?.map((tag) => (
-              <li key={tag}>{tag}</li>
-            ))}
-          </ul>
+          <div className="mt-2">
+            <div className="mb-2 flex items-center justify-between border-b">
+              <div className="hidden text-xl md:block">{t('filter-tasks')}</div>
+              <div className="text-2xl md:hidden">{t('your-retirement-checklist')}</div>
+              <IconButton
+                color="primary"
+                onClick={() => setExpanded(!expanded)}
+                aria-expanded={expanded}
+                aria-label={t('show-filters')}
+              >
+                <ExpandMore className="hidden md:block" />
+                <FilterList className="h-10 w-10 rounded-full bg-[#008490] p-1 text-white hover:bg-[#00545f] md:hidden" />
+              </IconButton>
+            </div>
+            <Collapse in={expanded}>
+              <FormGroup className="space-y-2" onChange={handleChange}>
+                {['life-in-retirement', 'managing-finances', 'public-pensions', 'other-benefits', 'time-sensitive'].map(
+                  (tag) => (
+                    <FormControlLabel
+                      key={tag}
+                      className="m-0 rounded-full bg-gray-200"
+                      control={
+                        <Checkbox
+                          value={tag}
+                          icon={<RadioButtonUncheckedIcon />}
+                          checkedIcon={<RadioButtonCheckedIcon />}
+                        />
+                      }
+                      label={t(tag)}
+                    />
+                  )
+                )}
+              </FormGroup>
+            </Collapse>
+          </div>
         </section>
         <section id="content" className="lg:col-span-8 xl:col-span-9">
           <NestedAccordion
             linksHeader={t('links-header')}
             sectionTitle={beforeRetiring.title}
             subSectionTitle={beforeRetiring.subTitle}
-            tasks={beforeRetiring.tasks}
+            tasks={beforeRetiring.tasks.filter((task) => filterTasksByTag(task, { tags: tagsToFilter }))}
```

Comment:
The filtering is done server side. Update the url's `filters` param on task filter change. Filters is pass to the component, update the object, convert it to base64 url-safe then router push the new url. GetServiceSideProps should be good handle the filtering.

Reply:
I've pushed a commit to your branch with the implemented solution. I was playing with it and implemented it at the same time.